tests/WriteDefaultsTests.js

```diff
@@ -0,0 +1,198 @@
+                listener: "gpii.test.nexus.verifyReadDefaultsResponse",
+                args: ["{arguments}.0", "{readDefaultsAgainRequest}", ["fluid.component", "gpii.tests.nexus.writeDefaults.newGrade"]]
+            }
+
+            // TODO: Update the grade definition and verify
+
+        ]
+    },
+    {
+        name: "Write Defaults with badly formed JSON",
+        gradeNames: "gpii.test.nexus.testCaseHolder",
+        expect: 3,
+        config: {
+            configName: "gpii.tests.nexus.config",
+            configPath: "%gpii-nexus/tests/configs"
+        },
+        testGradeName: "gpii.tests.nexus.writeDefaults.badlyFormedJson",
+        sequence: [
+            {
+                func: "{writeDefaultsRequest}.send",
+                args: [
+                    "{",
+                    {
+                        headers: {
+                            "Content-Type": "application/json"
+                        }
+                    }
+                ]
+            },
+            {
+                event: "{writeDefaultsRequest}.events.onComplete",
+                listener: "kettle.test.assertErrorResponse",
+                args: [{
+                    message: "Write Defaults returns 400 for badly formed JSON",
+                    errorTexts: "Unexpected end of input",
+                    string: "{arguments}.0",
+                    request: "{writeDefaultsRequest}",
+                    statusCode: 400
+                }]
+            }
+        ]
+    }
+];
+
+
+// TODO: GPII-1542 Test sending a badly-formed grade definition (such as bad compact invoker)
+
+// To implement testing of badly-formed grade definition, we will need
+// to replace the default Kettle run-time and test-time fluid.fail()
+// error handling. The default fail handlers cause the current Kettle
+// request to be rejected and the current test to be failed.
+//
+// Fail handlers are managed with Infusion event listeners. Each
+// namespaced listener has a stack of handlers -- the one at the top
+// is active. If the stack is popped, the top one is removed and the
+// next below becomes active.
+//
+// For the Kettle run-time Write Defaults request handler, we will:
+//
+// - addListener to failureEvent.fail with function fluid.identity
+//   (push a NOP handler onto the top of the stack)
+//   (see kettle.test.pushInstrumentedErrors)
+// - try fluid.defaults()
+// - catch exception, if instanceof fluid.FluidError, 400, otherwise 500
+// - finally removeListener failureEvent.fail (revert back to default handler)
+//
+// And similarly for the test-time fail handler.
+//
+// See:
+//
+// - https://github.com/amb26/kettle/blob/KETTLE-32/tests/ErrorTests.js#L220
+// - https://github.com/amb26/kettle/blob/KETTLE-32/lib/test/KettleTestUtils.js#L87
+// - https://github.com/fluid-project/infusion/blob/master/tests/test-core/jqUnit/js/jqUnit.js#L264
+
+/*
+gpii.tests.nexus.writeDefaults.badlyFormedInvokerGradeOptions = {
```

Comment:
Yes, that's my thinking also. My focus this week will be to get Nexus ready to move to trunk.